### PR TITLE
Show more honest errors when a wildcard name is used in implicit0

### PR DESCRIPTION
**Pull request by kubukoz**
Apparently the `if t != termNames.WILDCARD` check would never be true (the tree differs a bit in that the whole Bind node doesn't appear when a wildcard name is used), so this PR makes this check actually match when a wildcard is used with or without a type.

### CODE REVIEW

**Comment by kubukoz** [comment]
BTW, this repo apears no to have any CI set up, do you think some initial Travis setup would be helpful?

File: better-monadic-for/src/main/scala/com/olegpy/bm4/ImplicitPatterns.scala [hunk 1]
```diff
@@ -102,19 +102,43 @@ trait ImplicitPatterns extends TreeUtils { self =>
...
+        val andType = if(typed) " and type" else ""
+
+        reporter.error(
+          arg.pos,
+          s"implicit pattern requires an identifier, but a wildcard was used: `$t`. " +
+            s"This doesn't introduce anything into the implicit scope. You might want to remove the implicit0 pattern$andType."
+        )
+        false
+
+      case q"implicit0(${Bind(t: TermName, Typed(Ident(termNames.WILDCARD), _))})" =>
```
Comment (kubukoz) [inline better-monadic-for/src/main/scala/com/olegpy/bm4/ImplicitPatterns.scala:115]: Do you know why this `Ident(termNames.WILDCARD)` pattern is there? Some synthetic stuff made up earlier by the compiler, I guess?

btw. is there some tool for quickly getting a structure (similar to these patterns) from a compiler tree? All I relied on was `toString` and `summaryString`...

Comment (oleg-py) [reply better-monadic-for/src/main/scala/com/olegpy/bm4/ImplicitPatterns.scala:115]: IIRC the desugared trees are always transformed in this way:
`case a` => `case a @ _`
`case a: Thing` => `case a @ (_: Thing)`

There should be `showRaw` on the reflection universe which would do exactly what you're asking. It works even with runtime universe (so you can try it out in repl/worksheet) but the output is horrendous to read 😅

**Comment by oleg-py** [comment]
I considered setting CI up, but right now the testing also involves me checking manually the warnings being generated (which I don't know how to automate, testing libs that use macros to check for compile errors didn't use the plugin), so it wasn't really of a significant benefit.

**Comment by kubukoz** [comment]
Well you can test the PC output for errors :D

**Comment by oleg-py** [comment]
That's not a good way forward, given it's effectively EOL'd and will probably not be updated for 2.13 and up

**Comment by kubukoz** [comment]
Oh, I didn't know. We need other ways, then...

**Pull request merged**